### DISCUSSION TITLE
code styling script crashing

### DISCUSSION THREAD

**Discussion by Adrian-Lock** [MetOffice/simulation-systems#103] [UM]
I suspect this is just me but when I try to run the styling script on my branch (ie ". admin/code_styling/apply_styling" from the top branch directory) it gets as far as printing the initial messages (Something like "About to apply styling script" and lists 2 or 3 directories) before killing my terminal window (hence why I can't just cut-and-paste the output I do get!).  It also doesn't apply the styling fixes, of course.   Has anyone else ever encountered this or have any ideas how I can stop this happening, or even work out what the problem is?  I first had this problem on Friday, after many years of happily using the script, and it hasn't magically fixed itself by this morning 😪

**Comment by yaswant** [comment]
@Adrian-Lock you could try executing the command in debug mode in a subshell like `(set -x; .../apply_styling)` that way your parent shell wont get terminated.

OR, please check the file permission for `apply_styling` script; if it is executable, then a direct execution, instead of sourcing (`.`) the script, should not terminate the shell when `exit` is encountered.

**Reply by Adrian-Lock** [reply]
Thanks Yash.  I should have said that it mysteriously started working again a few months back, but useful to know should it mysteriously fail again!
